connector/src/main/java/com/google/cloud/hive/bigquery/connector/utils/avro/AvroUtils.java

```diff
@@ -71,6 +79,42 @@ public static Schema parseSchema(String jsonAvroSchema) {
     }
   }
 
+  /** Extract the Avro schema from the given table properties. */
+  public static Schema extractAvroSchema(Configuration conf, Properties tableProperties) {
+    String columnNameProperty = tableProperties.getProperty(serdeConstants.LIST_COLUMNS);
+    String columnTypeProperty = tableProperties.getProperty(serdeConstants.LIST_COLUMN_TYPES);
+    String columnCommentProperty = tableProperties.getProperty("columns.comments", "");
+    String columnNameDelimiter =
+        tableProperties.containsKey(serdeConstants.COLUMN_NAME_DELIMITER)
+            ? tableProperties.getProperty(serdeConstants.COLUMN_NAME_DELIMITER)
+            : String.valueOf(',');
+    boolean hasExternalSchema =
+        tableProperties.getProperty(AvroSerdeUtils.AvroTableProperties.SCHEMA_LITERAL.getPropName())
```

Comment:
Can we extract this out in a separate function and add a comment? The logic is a bit cumbersome